packages/nimble-components/src/dialog/index.ts

```diff
@@ -125,11 +124,33 @@ export class Dialog<CloseReason = void> extends FoundationElement {
         if (this.preventDismiss) {
             event.preventDefault();
         } else {
-            this.resolveShow!(UserDismissed);
-            this.resolveShow = undefined;
+            this.notifyClosed(UserDismissed);
         }
         return true;
     }
+
+    /**
+     * @internal
+     */
+    public closeHandler(): void {
+        if (this.resolveShow) {
+            // If
+            // - the browser implements dialogs with the CloseWatcher API, and
+            // - the user presses ESC without first interacting with the dialog (e.g. clicking, scrolling),
+            // the cancel event is not fired and the dialog just closes.
```

Comment:
the cancel event is not fired but the close event still is? If that's what's described can that be clarified?

```suggestion
            // the cancel event is not fired, but the close event still is, and the dialog just closes.
```